### PR TITLE
Multiversionize remaining poetry deps

### PR DESCRIPTION
**Pull request by justinvreeland**

### CODE REVIEW

**Comment by justinvreeland** [comment]
Needs cache control to be rebuilt https://github.com/wolfi-dev/os/pull/28246

**Comment by justinvreeland** [comment]
another missing rebuild: https://github.com/wolfi-dev/os/pull/28257

**Comment by justinvreeland** [comment]
Our automation doesn't detect runtime deps in the same PR correctly it seems so this will wait to be rebased on https://github.com/wolfi-dev/os/pull/28322

**Comment by TimKnight-DWP** [comment]
@justinvreeland - I am looking into building an apko image with poetry

I want to do something _like_

```
contents:
  keyring:
    - https://packages.wolfi.dev/os/wolfi-signing.rsa.pub
  repositories:
    - https://packages.wolfi.dev/os
  packages:
    - wolfi-base
    - python-3.11
    - py3.11-poetry
 ```

i.e. install python-3.11 and then install poetry for python 3.11 -> right now when I do
```
contents:
  keyring:
    - https://packages.wolfi.dev/os/wolfi-signing.rsa.pub
  repositories:
    - https://packages.wolfi.dev/os
  packages:
    - wolfi-base
    - python-3.11
    - poetry
 ```
Poetry pulls all it's transitive dependencies from py3.12 

Are these changes you're working on intended to allow me to do what I'm intending?

**Comment by justinvreeland** [comment]
> @justinvreeland - I am looking into building an apko image with poetry
> 
> I want to do something _like_
> 
> ```
> contents:
>   keyring:
>     - https://packages.wolfi.dev/os/wolfi-signing.rsa.pub
>   repositories:
>     - https://packages.wolfi.dev/os
>   packages:
>     - wolfi-base
>     - python-3.11
>     - py3.11-poetry
> ```
> 
> i.e. install python-3.11 and then install poetry for python 3.11
> 
> The current valid apko syntax for that is (as py3.11-poetry doesn't exist in wolfi os):
> 
> ```
> contents:
>   keyring:
>     - https://packages.wolfi.dev/os/wolfi-signing.rsa.pub
>   repositories:
>     - https://packages.wolfi.dev/os
>   packages:
>     - wolfi-base
>     - python-3.11
>     - poetry
> ```
> 
> Poetry pulls all it's transitive dependencies from py3.12
> 
> Are these changes you're working on intended to allow me to do what I'm intending?

These changes are for some of the poetry deps the changes you're looking for will be landing in: https://github.com/wolfi-dev/os/pull/28255 once the remaining deps are fixed.

**Comment by TimKnight-DWP** [comment]
@justinvreeland perfect thank you

**Comment by smoser** [comment]
To be clear, @TimKnight-DWP , what you'll end up with is the ability to do:

```yaml
contents:
..
  packages:
    - wolfi-base
    - py3.11-poetry
```

You won't have to declare that you need python-3.11as py3.11-poetry knows it depends on that.
Its not a "install python-3.11 _and then_ install poetry for python 3.11", but rather "install the poetry built for python 3.11 and its dependencies".

File: py3-keyring.yaml [hunk 1]
```diff
@@ -2,30 +2,33 @@
...
+  epoch: 1
   description: Store and access your passwords safely.
   copyright:
     - license: "MIT"
   dependencies:
-    runtime:
-      - py3-jaraco.classes
-      - py3-importlib-metadata
-      - py3-importlib-resources
-      - py3-SecretStorage
```
Comment (smoser) [inline py3-keyring.yaml:14]: this was wierd. hmm. case?

Comment (justinvreeland) [reply py3-keyring.yaml:14]: yeah it came form an auto conversion of the package. The upstream stylizes it that way for the Name but not the package file.   I'll remove the provides for this too once everything lands and no-one needs the dep any more.

**Review by smoser — APPROVED** [review]

**Pull request merged**